example/CMakeLists.txt

```diff
@@ -32,7 +32,9 @@ endif()
 # Find the needed BlockFactory components:
 # - BlockFactoryCore contains the core classes such as Block and Signal
 # - BlockFactoryMex is required at runtime for loading the library from Simulink
-find_package(BlockFactory 1 REQUIRED COMPONENTS BlockFactoryCore BlockFactoryMex)
+find_package(BlockFactory
+    REQUIRED COMPONENTS BlockFactoryCore
+    OPTIONAL_COMPONENTS BlockFactoryMex)
```

Comment:
Just to understand, what is the point of searching for `BlockFactoryMex` in this case? We should check if it is there if Matlab is found/enabled?

Reply:
As written in the comment in the code, it is a runtime dependency. The example compiles even if it is not found. I put it there when I wrote the tutorial to let the reader understand this concept.